src/replay.cpp

```diff
@@ -705,42 +705,42 @@ REPLAY_RETURN do_replay_handle(bool one_move)
 			ERR_REPLAY << "found "<<  cfg->debug() <<" in replay" << std::endl;
 			//do nothing
 		}
-		else if (const config &child = cfg->child("speak"))
+		else if (const config &speak = cfg->child("speak"))
 		{
-			const std::string &team_name = child["team_name"];
-			const std::string &speaker_name = child["id"];
-			const std::string &message = child["message"];
+			const std::string &team_name = speak["team_name"];
+			const std::string &speaker_name = speak["id"];
+			const std::string &message = speak["message"];
 			//if (!preferences::parse_should_show_lobby_join(speaker_name, message)) return;
 			bool is_whisper = (speaker_name.find("whisper: ") == 0);
 			if(resources::recorder->add_chat_message_location()) {
 				DBG_REPLAY << "tried to add a chat message twice.\n";
 				if (!resources::controller->is_skipping_replay() || is_whisper) {
-					int side = child["side"];
-					resources::screen->get_chat_manager().add_chat_message(get_time(child), speaker_name, side, message,
+					int side = speak["side"];
+					resources::screen->get_chat_manager().add_chat_message(get_time(speak), speaker_name, side, message,
 						(team_name.empty() ? events::chat_handler::MESSAGE_PUBLIC
 						: events::chat_handler::MESSAGE_PRIVATE),
 						preferences::message_bell());
 				}
 			}
 		}
-		else if (const config &child = cfg->child("label"))
+		else if (const config &label = cfg->child("label"))
 		{
-			terrain_label label(resources::screen->labels(), child);
+			terrain_label terrain_label(resources::screen->labels(), label);
```

Comment:
Instead of renaming `label` to `terrain_label`, I suggest renaming `child` above to `label_data` or `label_config`.

(The reason for my suggestion is that this variable is the real label, whereas the above one is merely configuration for creating the label.)